### DISCUSSION TITLE
Convergence Problem of TDS

### DISCUSSION THREAD

**Discussion by yzhuold** [CURENT/andes#413] [Q&A] [answered]
Dear Dr. Cui and other experts,

I want to simulate load shedding after fault clearance in the IEEE14 bus system. A three-phase fault occurred on bus 3 at 0.1s and was cleared at 0.2s. Line 3 also tripped at 0.2s. Load shedding is implemented at 0.3s by altering the Ppf value of [PQ.]([ieee14.xlsx](https://github.com/cuihantao/andes/files/10821094/ieee14.xlsx))
[ieee14.xlsx](https://github.com/cuihantao/andes/files/10821103/ieee14.xlsx)

In order to implement load shedding, I need to set p2p and q2q to 1 before PFlow and TDS so that I can alter Ppf to change the load of PQ. But when conducting TDS, there is always a problem that shows “Convergence is not likely”. TDS runs well without setting p2p and q2q to 1. Could you please give me some advice on it? 
![image](https://user-images.githubusercontent.com/120257504/221103785-6c47cb2c-c8f3-4897-89cd-65011c77e965.png)

ss=andes.load('ieee14.xlsx')
ss.PQ.config.p2p=1
ss.PQ.config.p2i=0
ss.PQ.config.p2z=0
ss.PQ.config.q2q=1
ss.PQ.config.q2i=0
ss.PQ.config.q2z=0
ss.PQ.pq2z=0
ss.PFlow.run()
ss.TDS.config.tf=3
ss.TDS.config.criteria = 0 
ss.TDS.run()

Kind Regards
yang

**Comment by cuihantao** [comment, marked as answer]
Hello Yang,

This is very likely that the power flow does not have a solution when the load model is constant power. The fault is drawing too much power from the network, so that the constant power constaints on the loads cannot be met due to voltage drop. 

One way is to increase the fault impedance, i.e., the `xf` of the Fault. This will create a fault drawing fewer reactive power. You might tune it to find a feasible solution, then reduce the load and reduce the fault impedance until that meets your target.

**Reply by yzhuold** [reply]
Dear Dr. Cui,

Thanks for your reply! The solution works. I will try to tune the impedance to find an appropriate value. Thanks a lot.

By the way, can you share the "IEEE39.xlsx" file which cintains dynamic datas? I always have problems when I convert  "IEEE39.raw" and "IEEE39.dyr" to "IEEE39.xlsx" due to some mismatch of models. It will be better if you can share both "raw, dyr" and "xlsx" files.

Wish you a happy weekend.

Kind Regards
yang

**Reply by cuihantao** [reply]
Hello, maybe @jinningwang can help. I don't have it at hand and will need to look it up tomorrow.

**Reply by jinningwang** [reply]
> Hello, maybe @jinningwang can help. I don't have it at hand and will need to look it up tomorrow.

I can open a PR to include the ieee39 cases with dynamic devices. Is it ok?

Regards,
Jinning

**Reply by cuihantao** [reply]
Absolutely!

**Reply by yzhuold** [reply]
Thank you so much for sharing!

**Reply by jinningwang** [reply]
Hi Yang,

I opened a [PR](https://github.com/cuihantao/andes/pull/414) to include the IEEE 39-bus case in ANDES xlsx format with dynamic devices. It will be merged into develop branch or you can directly grab the case from PR.
I do not have the ``dyr`` file at hand, we manually added the dynamic devices into the case converted from the ``raw`` file.

Regards,
Jinning

**Reply by yzhuold** [reply]
Hi @Jinning,

Many thanks! I have found the document you uploaded.

Kind Regards
yang

**Comment by jinningwang** [comment]
Hello @yzhuold,

Please ***MARK OUT*** an answer to your question if you find any reply appropriate. In this way, the Q&A will be a valuable input to this community that can be helpful for future users.
We commit efforts to maintain this open-source software. We will be grateful if you can also contribute to it if our work really helps you somewhere.

Regards,
Jinning
